src/main/java/com/google/sps/servlets/LoadInterviewsServlet.java

```diff
@@ -0,0 +1,276 @@
+    this.availabilityDao = availabilityDao;
+    this.scheduledInterviewDao = scheduledInterviewDao;
+    this.instant = instant;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int timezoneOffsetMinutes = Integer.parseInt(request.getParameter("timeZoneOffset"));
+    Preconditions.checkArgument(
+        Math.abs(timezoneOffsetMinutes) <= 720,
+        "Offset greater than 720 minutes (12 hours): %s",
+        timezoneOffsetMinutes);
+    ZoneOffset timezoneOffset = convertIntToOffset(timezoneOffsetMinutes);
+    ZonedDateTime day = generateDay(instant, timezoneOffset);
+    ZonedDateTime utcTime = day.withZoneSameInstant(ZoneOffset.UTC);
+    // TODO: Decide how precise this range should be. Must start after or at now, when should end
+    // be?
+    Instant startOfRange = utcTime.toInstant();
+    Instant endOfRange = utcTime.toInstant().plus(6, ChronoUnit.DAYS);
+    List<Availability> availabilitiesInRange =
+        availabilityDao.getInRangeForAll(startOfRange, endOfRange);
+    List<PossibleInterviewSlot> possibleInterviews =
+        getPossibleInterviewSlots(availabilitiesInRange, startOfRange, endOfRange, timezoneOffset);
+    Set<String> dates = new HashSet<String>();
+    for (PossibleInterviewSlot possibleInterview : possibleInterviews) {
+      dates.add(possibleInterview.date());
+    }
+
+    List<ArrayList<PossibleInterviewSlot>> possibleInterviewsForWeek =
+        new ArrayList<ArrayList<PossibleInterviewSlot>>();
+    for (String date : dates) {
+      ArrayList<PossibleInterviewSlot> dayOfSlots = new ArrayList<PossibleInterviewSlot>();
+      for (PossibleInterviewSlot possibleInterview : possibleInterviews) {
+        if (date.equals(possibleInterview.date())) {
+          dayOfSlots.add(possibleInterview);
+        }
+      }
+      sortInterviews(dayOfSlots);
+      possibleInterviewsForWeek.add(dayOfSlots);
+    }
+    sortWeek(possibleInterviewsForWeek);
+    request.setAttribute("weekList", possibleInterviewsForWeek);
+    RequestDispatcher rd = request.getRequestDispatcher("/possibleInterviewTimes.jsp");
+
+    try {
+      rd.forward(request, response);
+    } catch (ServletException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  // Uses an Instant and a timezoneOffset to create a ZonedDateTime instance.
+  private static ZonedDateTime generateDay(Instant instant, ZoneOffset timezoneOffset) {
+    return instant.atZone(ZoneId.ofOffset("UTC", timezoneOffset));
+  }
+
+  // This method takes the timezoneOffsetMinutes int and converts it
+  // into a proper ZoneOffset instance.
+  private static ZoneOffset convertIntToOffset(int timezoneOffsetMinutes) {
+    return ZoneOffset.ofHoursMinutes((timezoneOffsetMinutes / 60), (timezoneOffsetMinutes % 60));
+  }
+
+  private List<PossibleInterviewSlot> getPossibleInterviewSlots(
+      List<Availability> allAvailabilities,
+      Instant startOfRange,
+      Instant endOfRange,
```

Comment:
To minimize params, could we replace this with a TimeRange instance?

Reply:
Done